projecttacoma/fqm-testify
# Add search bar to search data elements

# Summary
Adds a search bar for the user to filter the data elements that appear in the test resources display.

## New Behavior
The user can now use a search bar located above the test resources display to search by any criteria that is displayed (the data type, the ValueSet name, and ValueSet URL). As the user types, the display of available data elements gets filtered using fuzzy searching. If the user does not type anything in the search bar, all the data elements will be shown in the display.

## Code Changes
* Brings in the FuseJS dependency
* Creates new `SearchBar` component
* Adds recoil state for the search query that the user types into the search bar
* Adds `useEffect()` that listens for changes to the user’s search query - when it changes, uses a Fuse object to search the data and filter the results
* Adds state in `TestResourcesDisplay` for the data requirements that will be displayed based on the results of the search

# Testing Guidance
* Run unit tests (checking for presence of search bar + fuzzy searching capabilities)
* Import a measure bundle, create a patient, and look at the test resources display.
* Try out a few searches (resource type, ValueSet name, ValueSet URL) and check that the results get filtered. (Feel free to gauge how “well” you think the search does and leave feedback about what types of data we should search on if you have any suggestions)
* Try backspacing to clear the search bar and check that all the data elements have returned to the display
